src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Create/CreateResourceHandler.cs

```diff
@@ -34,8 +34,11 @@ public async Task<UpsertResourceResponse> Handle(CreateResourceRequest message,
 
             var resource = message.Resource.Instance.ToPoco<Resource>();
 
-            // If an Id is supplied on create it should be removed/ignored
-            resource.Id = null;
+            // Check to see if we want to persist the supplied Id
+            if (!message.PersistId)
```

Comment:
Can we just use `UpsertResourceRequest` instead in this case?

Reply:
We possibly could, but I think we may get into some cases where we'd get unexpected behavior. 

A concrete example that looks like would already exists is that in `R4` it looks like the `UpsertResourceHandler` does a check for a setting called `RequireETag`. If that setting was enabled, you wouldn't be able to create in a transaction. We obviously could get around this by doing an additional check, but I'm not sure that end state is better

Reply:
This was updated to using a settable func so that we're only modifying how create/upsert creates a new id.